internal/cache/llmcache/llm_cache.go

```
@@ -0,0 +1,380 @@
+
+// BuildKey constructs the canonical, tenant-scoped cache key.
+//
+//	kc:llm:{tenant_id}:{task_type}:{xxhash64(keyParts...)}
+//
+// A NUL byte separates key parts to prevent cross-field hash collisions
+// (e.g. ("ab","c") vs ("a","bc")). The engine never assumes anything about the
+// business meaning of keyParts — that is the caller's responsibility.
+func BuildKey(tenantID, taskType string, keyParts ...string) string {
+	h := xxhash.New()
+	for _, p := range keyParts {
+		h.WriteString(p)
```

golangci-lint (2.12.2)

[error] Error return value of `h.WriteString` is not checked

(errcheck)